### PR TITLE
Use CRD for state persistence

### PR DESCRIPTION
**Pull request by r-dilip**

### CODE REVIEW

File: source/code/plugin/health/cluster_health_state.rb [hunk 1]
```diff
@@ -0,0 +1,117 @@
+require "net/http"
+require "net/https"
+require "uri"
+
+module HealthModel
+    class ClusterHealthState
+
+        attr_reader :token_file_path, :cert_file_path, :log, :http_client, :uri, :token
+        @@resource_uri_template = "%{kube_api_server_url}/apis/azmon.container.insights/v1/namespaces/kube-system/healthstates/chs"
```
Comment (vishiy) [inline source/code/plugin/health/cluster_health_state.rb]: chs ??

File: source/code/plugin/health/cluster_health_state.rb [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            if !File.exist?(@cert_file_path)
+              raise "#{@cert_file_path} doesnt exist"
+            else
+              http.ca_file = @cert_file_path
+            end
+            http.verify_mode = OpenSSL::SSL::VERIFY_PEER
+            return http
+        end
+
+        def get_kube_api_server_url
```
Comment (vishiy) [inline source/code/plugin/health/cluster_health_state.rb]: why is this function duplicated (from kubeapiclient) ?

File: installer/datafiles/base_container.data [hunk 3]
```diff
@@ -125,8 +125,9 @@ MAINTAINER:              'Microsoft Corporation'
 
 
 /opt/microsoft/omsagent/plugin/health/aggregate_monitor.rb;                                         source/code/plugin/health/aggregate_monitor.rb; 644; root; root
-/opt/microsoft/omsagent/plugin/health/aggregate_monitor_instance_id_labels.rb;                      source/code/plugin/health/aggregate_monitor_instance_id_labels.rb; 644; root; root
+/opt/microsoft/omsagent/plugin/health/agg_monitor_id_labels.rb;                      		    source/code/plugin/health/agg_monitor_id_labels.rb; 644; root; root
```
Comment (vishiy) [inline installer/datafiles/base_container.data]: is this is what was causing the issue, this does not look that long. Are we sure the root cause is this ?

File: source/code/plugin/health/cluster_health_state.rb (hunk 1 above)
Comment (r-dilip) [reply source/code/plugin/health/cluster_health_state.rb]: cluster-health-state

File: installer/datafiles/base_container.data (hunk 3 above)
Comment (r-dilip) [reply installer/datafiles/base_container.data]: yeah. 100% positive. only change was the change to the filename and the agent started working on newer kernels

File: source/code/plugin/health/cluster_health_state.rb (hunk 2 above)
Comment (r-dilip) [reply source/code/plugin/health/cluster_health_state.rb]: that constructs the entire URL based on resource which technically can construct this. but then it raises a hairy mess of dependency with oms_common, and I cannot unit test it locally.

**Pull request merged**